### DISCUSSION TITLE
Change testing structure.

### DISCUSSION THREAD

**Discussion by jimmywarting** [agershun/alasql#1598] [General]
When i have been writing test for cross env. then i have mostly opt'ed out of using barely any testing frameworks or depend on any built in features. (such as [node:assert](https://nodejs.org/dist/latest-v19.x/docs/api/assert.html), mocka, global `describe` or `it` or anything really)

I just mostly create my own assertion functions that i need and most of the time it's just a simple function that checks if first param is truthy and if it isn't then it throw an error with the 2nd string parameter... And use that basically for everything.

Then i write my own test objects and push them into some storage.
it basically goes like:

```js
export const testCases = []

export default function assert(condition, message = '') {
  if (condition) return

  // Condition not passed, Throw an error
  throw new Error(message)
}

export function t(obj) {
  testCases.push(obj)
}
```
```js
import { assert, t } from '../xyz.js'

t({
  name: 'leaky test',
  skip: false, // optional (default to false)
  only: false, // optional (default to false)
  timeout: 3000, // optional
  async fn() {
    assert((1 + 1) === 2, 'one plus one should be two')
  }
})
```
Then depending on if i'm running things in Deno, or NodeJS then i use their own built in test runner.
and basically do something in lines of:

```js
// Deno test runner
import { testCases } from '../xyz.js'

for (const obj of testCases) {
  Deno.test(obj)
}
```

```js
// Node test runner
import test from 'node:test'
import { testCases } from '../xyz.js'

for (const { name, fn, ...opts } of testCases) {
  test(name, opts, fn)
}
```

So you could say that my way of writing test cases are more or less Isomorphic and can be executed with any test-runner i wish to use. weather it's NodeJS or Deno's built in test runner or mocha in browsers.

What do you think about this?

**Comment by mathiasrw** [comment]
If you dont pick a framework you end up creating your own. 

I dont feel this is the right project to start brewing our own cross platform testing framework. There must be something out there already. 

Mocha should have what we are looking for. It supports both node, browser and deno via ESM. Check https://dev.to/craigmorten/testing-your-deno-apps-with-mocha-4f35 for inspiration.
